_posts/2021-03-14-Julia-1.6-what-has-changed-since-1.0.md

```diff
@@ -0,0 +1,737 @@
+
+ - Reproducing an issue.
+ - Answering questions on Stack Overflow etc.
+ - Cross-checking the behavior of the last release of a package you currently have `dev`ed.
+ - Quickly trying out an idea.
+
+Remember that with Pkg, installing a version of a package you have installed before is incredibly fast.
+Pkg doesn't download it again, it basically just points to the existing version on disk.
+So using `pkg"activate --temp"` to quickly try something out is indeed quick.
+
+A more hacky use of it is after loading (via `using`) a package that you are working on, you can `pkg"activate --temp"` and install some of your test-time dependencies to reproduce a test failure without adding them to the main dependencies in the `Project.toml`.
+Though there is a better way if you use `test/Project.toml`.
+
+### Test Dependencies in their own Project.toml
+In Julia 1.0 and 1.1, [test specific dependencies are listed in the `[extras]` section](https://julialang.github.io/Pkg.jl/v1/creating-packages/#Test-specific-dependencies-in-Julia-1.0-and-1.1), and under `[targets] test=[...]`; with their compatibility listed in the main `[compat]`.
+It seemed like this might be extended for other things in the future, perhaps documentation.
+But it was found for documentation a separate `Project.toml` worked well (as long as you `pkg"dev .."` the parent package directory into its `Manifest.toml` so it uses the right version of the package it is documenting).
+
+The new `test/Project.toml` extends that idea.
+One part of that extension is to remove the need to `dev ..`, and other part is to make available all the main dependencies.
+It actually works on a different mechanism than the docs.
+It relies on stacked environments, which is a feature Julia has had since 1.0 via `LOAD_PATH`, but that is rarely used.
+
+One advantage of this is that you can activate that Project.toml, on top of the your existing environment by adding the test directory to your `LOAD_PATH`.
+I wrote some more details on exactly how to do that on [Discourse](
+https://discourse.julialang.org/t/activating-test-dependencies/48121/7?u=oxinabox).
+It feels kind of hacky, because it is.
+At some point there might be a nicer user interface for stacked environments like this.
+
+
+### BinaryBuilder, Artifacts, Yggdasil and JLL packages
+
+This story was beginning to be told around Julia 1.0 time,
+but it wasn't really complete nor built into Pkg until Julia 1.3.
+You can read the documentation on [Artifact](https://julialang.github.io/Pkg.jl/v1/artifacts/), [BinaryBuilder](https://github.com/JuliaPackaging/BinaryBuilder.jl), and [Yggdasil](https://github.com/JuliaPackaging/Yggdrasil) for full details.
+This is the story about how Julia works with binary dependencies.
+As of now, it works really well.
+
+Sometimes people misunderstand the claims about Julia solving the two language problem as saying everything should be rewritten in Julia.
+Which is far from the truth, ain't noone got time for that.
```

Comment:
the meme is "ain't nobody got time for that" but you call XD

Reply:
gotta get your memes right
```suggestion
Which is far from the truth, ain't nobody got time for that.
```